.github/workflows/quick.yaml

```diff
@@ -159,17 +169,19 @@ jobs:
 
   CodeQL-tests:
 
-    runs-on: [ ubuntu-22.04 ]
+    runs-on: [ ubuntu-24.04 ]
 
     permissions:
       security-events: write
 
     steps:
 
-      - name: Install Squid prerequisite Linux packages
+      - name: Install prerequisite Linux packages
         run: |
+          cp -p /etc/apt/sources.list.d/ubuntu.sources /tmp/ubuntu.sources.bak
           # required for "apt-get build-dep" to work
-          sudo sed --in-place -E 's/# (deb-src.*updates main)/  \1/g' /etc/apt/sources.list
+          sudo sed --in-place -E 's/^(Types: deb)$/\1 deb-src/' /etc/apt/sources.list.d/ubuntu.sources
+          diff -u /tmp/ubuntu.sources.bak /etc/apt/sources.list.d/ubuntu.sources || true
```

Comment:
This `diff` and its `/tmp` file look like debugging traces left in the final PR. They are not useful in the final change (following `apt` usage will display any relevant errors). Please remove.